src/openrct2/ride/Ride.cpp

```diff
@@ -134,7 +131,7 @@ size_t RideManager::size() const
     size_t count = 0;
     for (size_t i = 0; i < _endOfUsedRange; i++)
     {
-        if (!_rides[i].id.IsNull())
+        if (!GetGameState().Rides[i].id.IsNull())
```

Comment:
I think it would be better to get the game state outside of the loop and maybe even have a reference to Rides.